nixos/modules/services/security/esdm.nix

```diff
@@ -0,0 +1,137 @@
+{ lib, config, pkgs, ... }:
+
+let
+  cfg = config.services.esdm;
+  kernelVersion = config.boot.kernelPackages.kernel.version;
+  linux_6_3 = (lib.versionAtLeast kernelVersion "6.3");
+in
+{
+  options.services.esdm = {
+    enable = lib.mkEnableOption (lib.mdDoc "ESDM service configuration");
+    package = lib.mkPackageOptionMD pkgs "esdm" { };
+    kernelEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        enable ESDM kernel module
+      '';
+    };
+    serverEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        enable option for ESDM server service
+      '';
+    };
+    cuseRandomEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        enable option for ESDM cuse-random service
+      '';
+    };
+    cuseUrandomEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        enable option for ESDM cuse-urandom service
+      '';
+    };
+    procEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        enable option for ESDM proc service
+      '';
+    };
+    verbose = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        enable verbose ExecStart for ESDM
+      '';
+    };
+    esdmHashName = lib.mkOption {
+      type = lib.types.str;
+      default = "sha3-512";
+      description = lib.mdDoc ''
+        hash configration of the kernel module esdm_es
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    lib.mkMerge [
```

Comment:
Using config = mkMerge is causing inifinite recursions

Reply:
I am not sure under which exact conditions but it would be good to avoid this.

Reply:
If there is an issue we should definitively also warn users of it in the NixOS Manual because the manual "recommends" (as in it shows it in the example) `config = mkMerge` https://nixos.org/manual/nixos/stable/#sec-option-definitions-merging

Reply:
I don't think it's a real issue.